test/system/500-networking.bats

```diff
@@ -390,4 +390,89 @@ load helpers
     run_podman network rm -f $netname
 }
 
+# Test for https://github.com/containers/podman/issues/10052
+@test "podman network connect/disconnect with port forwarding" {
+    random_1=$(random_string 30)
+    HOST_PORT=12345
+    SERVER=http://127.0.0.1:$HOST_PORT
+
+    # Create a test file with random content
+    INDEX1=$PODMAN_TMPDIR/hello.txt
+    echo $random_1 > $INDEX1
+
+    local netname=testnet-$(random_string 10)
+    run_podman network create $netname
+    is "$output" ".*/cni/net.d/$netname.conflist" "output of 'network create'"
+
+    local netname2=testnet2-$(random_string 10)
+    run_podman network create $netname2
+    is "$output" ".*/cni/net.d/$netname2.conflist" "output of 'network create'"
+
+    # First, run a container in background to ensure that the rootless cni ns
+    # is not destroyed after network disconnect.
+    run_podman run -d --network $netname $IMAGE top
+    background_cid=$output
+
+    # Run a httpd container on first network with exposed port
+    run_podman run -d -p "$HOST_PORT:80" \
+            --network $netname \
+            -v $INDEX1:/var/www/index.txt:Z \
+            -w /var/www \
+            $IMAGE /bin/busybox-extras httpd -f -p 80
+    cid=$output
+
+    # Verify http contents: curl from localhost
+    run curl --connect-timeout 3 -s $SERVER/index.txt
```

Comment:
My mistake: `--connect-timeout` is a useless option. It just hangs forever, and the tests will eventually hang and time out. The correct option is `--max-time`.

Reply:
fixed